keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/constraints/ConstraintState.kt

```diff
@@ -26,6 +26,8 @@ data class ConstraintState(
   val deliveryConfigName: String,
   val environmentName: String,
   val artifactVersion: String,
```

Comment:
FYI this was nullable to allow artifact reference to be rolled out originally. Do any recent artifacts have this, or is it only a couple of old ones? The code should set reference to name if it's not passed in upon submission of the delivery config